nestkernel/conn_parameter.h

```diff
@@ -145,50 +145,82 @@ class ScalarIntegerParameter : public ConnParameter
 
 
 /**
- * Array parameter, returning values in order.
+ * Array parameter classes, returning values in order.
  *
  * - The array of values must not be empty
  *   (so return 0 for number_of_values can signal non-array parameter)
  * - Throws exception if more values requested than available.
  *
  */
-class ArrayParameter : public ConnParameter
+
+class ArrayDoubleParameter : public ConnParameter
 {
 public:
-  ArrayParameter( const std::vector< double >& values )
-    : values_( values )
-    , next_( values_.begin() )
+  ArrayDoubleParameter( const std::vector< double >& values, const size_t nthreads )
+    : values_( &values )
+    , next_( nthreads, values_->begin() )
```

Comment:
Can you write some documentation explaining the architecture of `ArrayDoubleParameter` and `ArrayIntegerParameter`.